### ISSUE TITLE
Are the labels even being used in the backend?

### ISSUE THREAD

**Issue by rohan1561** [teticio/llama-squad#4]
Hello! Thank you for the amazing work! I was working on something similar and realized that using the Trainer object (which the SFTTrainer sub-classes), the labels will be set to None if self.label_smoothing is None in the Trainer class of hf. Specifically I refer you to the following lines in the trainer.py file at hf source

```python
    def compute_loss(self, model, inputs, return_outputs=False):
        """
        How the loss is computed by Trainer. By default, all models return the loss in the first element.

        Subclass and override for custom behavior.
        """
        if self.label_smoother is not None and "labels" in inputs:
            labels = inputs.pop("labels")
        else:
            labels = None
        outputs = model(**inputs)    
```
I was just curious because when I printed out the labels here in my own problem, they were being set to None. I was wondering is that is also an issue with your implementation..
Thank you!

**Comment by teticio** [comment]
Thanks!

With Causal LM in HuggingFace, the labels are usually the input_ids shifted by 1 place. This basically implements teacher forcing (c.f., seq2seq models). The loss function is the cross entropy loss over the logits against the token specified by the label averaged over all the output tokens. If there is a -100 value in the label, the cross entropy is not calculated for this output token.

I don't think label smoothing will work out of the box for this. There is specific code for this kind of model - you'd have to dig around a bit more in the transformers code to find it. Hope this helps.
